apps/manager_app/auth_card.c

```diff
@@ -332,20 +346,33 @@ static manager_error_code_t handle_sign_card_serial(
   instruction_scr_init(auth_card_data->ctx.message,
                        auth_card_data->ctx.heading);
 
-  /* TODO: Move snippet to sign serial flow*/
-  {
-    nfc_en_select_card_task();
-
-    evt_status_t status = get_events(EVENT_CONFIG_NFC, TIMEOUT_SELECTION);
-
-    HANDLE_P0_EVENTS(status.p0_event);
-
-    // NFC event occurred
+  card_sign_data_config_t sign_serial_config =
+      INIT_SIGN_SERIAL_CONFIG(auth_card_data->ctx.acceptable_cards,
+                              auth_card_data->ctx.family_id,
+                              true,
+                              auth_card_data->ctx.heading,
+                              auth_card_data->ctx.message,
+                              resp->serial_signature.serial);
+  card_error_type_e status = card_sign_auth_data(&sign_serial_config);
+
+  switch (status) {
+    case CARD_OPERATION_SUCCESS:
+      memcpy(resp->serial_signature.signature,
+             sign_serial_config.signature,
+             sizeof(resp->serial_signature.signature));
+      if (DEFAULT_UINT32_IN_FLASH == U32_READ_BE_ARRAY(get_family_id())) {
+        set_family_id_flash(auth_card_data->ctx.family_id);
+      }
```

Comment:
We are already writing family ID in flash memory during "card tap check".. so are we deeming it unsafe to be there?

Reply:
removed